### ISSUE TITLE
Wrong function name used in examples/customized_models/HGT/hgt_nc.py

### ISSUE THREAD

**Issue by thvasilo** [awslabs/graphstorm#125]
Running `pylint --rcfile tests/lint/pylintrc --errors-only` on `examples/customized_models/HGT/hgt_nc.py` produces:

```
examples/customized_models/HGT/hgt_nc.py:332:17: E1101: Instance of 'GSgnnNodePredictionTrainer' has no 'get_best_model' member (no-member)
```

**Comment by thvasilo** [comment]
Other error warning from this file:

```
pylint --disable R,C --rcfile tests/lint/pylintrc examples/customized_models/HGT/hgt_nc.py
************* Module hgt_nc
examples/customized_models/HGT/hgt_nc.py:101:38: E1101: Module 'dgl.function' has no 'v_dot_u' member (no-member)
examples/customized_models/HGT/hgt_nc.py:112:38: E1101: Module 'dgl.function' has no 'u_mul_e' member (no-member)
examples/customized_models/HGT/hgt_nc.py:112:76: E1101: Module 'dgl.function' has no 'sum' member (no-member)
examples/customized_models/HGT/hgt_nc.py:117:16: W0105: String statement has no effect (pointless-string-statement)
examples/customized_models/HGT/hgt_nc.py:190:4: W0237: Parameter 'input_nodes' has been renamed to 'epoch' in overridden 'HGT.forward' method (arguments-renamed)
examples/customized_models/HGT/hgt_nc.py:190:42: W0613: Unused argument 'edge_feats' (unused-argument)
examples/customized_models/HGT/hgt_nc.py:190:62: W0613: Unused argument 'epoch' (unused-argument)
examples/customized_models/HGT/hgt_nc.py:190:72: W0613: Unused argument 'total_steps' (unused-argument)
examples/customized_models/HGT/hgt_nc.py:217:4: W0221: Number of parameters was 5 in 'GSgnnNodeModelInterface.predict' and is now 4 in overridden 'HGT.predict' method (arguments-differ)
examples/customized_models/HGT/hgt_nc.py:246:9: W0621: Redefining name 'args' from outer scope (line 365) (redefined-outer-name)
examples/customized_models/HGT/hgt_nc.py:332:17: E1101: Instance of 'GSgnnNodePredictionTrainer' has no 'get_best_model' member (no-member)

------------------------------------------------------------------
Your code has been rated at 8.69/10 (previous run: 8.01/10, +0.68)
```

Note especially `W0221: Number of parameters was 5 in 'GSgnnNodeModelInterface.predict' and is now 4 in overridden 'HGT.predict' method (arguments-differ)` and `W0237: Parameter 'input_nodes' has been renamed to 'epoch' in overridden 'HGT.forward' method (arguments-renamed)` which are also breaking likely.

Are our examples tested in e2e_tests or regression tests @classicsong ?

**Comment by zhjwy9343** [comment]
I don't think we test this example HGT model in end-2-end test. Instead we test a simple dummy customer codes in the tests/end2end-tests/customer-gnn/train.py, which has the same lint problem.

**Comment by classicsong** [comment]
For the lint errors related to DGL, it is because you do not install dgl locally.

**Comment by zhjwy9343** [comment]
The problem is related to the best model extraction mechanism. Currently GraphStorm use best_model_path to restore models. So this example code need to update to follow this new mechanism.

This issue had been fixed by the PR #172
